toshi-magento-plugin/Shipping/view/frontend/web/js/toshi.js

```diff
@@ -0,0 +1,232 @@
+                setAddress();
+                
+                jQuery(document).off('change', 'input[name=firstname]');
+                jQuery(document).on('change', 'input[name=firstname]', function() {
+                    modal.setFirstName(jQuery('input[name=firstname]').val());
+                });
+                
+                jQuery(document).off('change', 'input[name=lasttname]');
+                jQuery(document).on('change', 'input[name=lastname]', function() {
+                    modal.setLastName(jQuery('input[name=lastname]').val());
+                });
+                
+                jQuery(document).off('change', '#customer-email');
+                jQuery(document).on('change', '#customer-email', function() {
+                    modal.setEmail(jQuery('#customer-email').val());
+                });
+                
+                jQuery(document).off('change', 'input[name=telephone]');
+                jQuery(document).on('change', 'input[name=telephone]', function() {
+                    modal.setPhone(jQuery('input[name=telephone]').val());
+                });
+                
+                jQuery(document).off('change', addressFields);
+                jQuery(document).on('change', addressFields, function() {
+                    setAddress();
+                });
+                
+            }
+        }, 750);
+        
+        function setAddress() {
+            modal.setAddress({
+                addressLine1: jQuery('input[name=street\\[0\\]]').val(),
+                addressLine2: jQuery('input[name=street\\[1\\]]').val(),
+                town: jQuery('input[name=city]').val(),
+                province: jQuery('input[name=region]').val(),
+                postcode: jQuery('input[name=postcode]').val(),
+                country: jQuery('input[name=country_id]').val()
+            });
+        }
+        
+        const createProduct = (name, sku, qty, imageUrl, retailPrice, size, colour, availableSizes, availabilityType, availabilityDate) => {
+            return {
+                /** Mandatory Properties */
+                name: name,
+                size: size,
+                sku: sku,
+                quantity: qty,
+                imageUrl: imageUrl,
+                retailPrice: retailPrice,
+                finalPrice: retailPrice,
+
+                /** Optional Properties */
+                colour: colour,
+                availableSizes: availableSizes,
+
+                availabilityType: availabilityType,
+                availabilityDate: availabilityDate,
+            };
+        };
+
+        let products = [];
+        let sizeAttribute = checkoutConfig.toshiSizeAttribute;
+        let colorAttribute = checkoutConfig.toshiColorAttribute;
+
+        checkoutConfig.quoteItemData.forEach((item, index) => {
+            if (item.product_type === 'configurable') {
+                availableSizes = checkoutConfig.toshiData.products[index].additionalSizes;
```

Comment:
Ideally, this should just be the size up and the size down of the product. 

E.g. If the customer selected M, the array should containt `['S', 'L']`

If the customer selected S then the array should contain `['S']`

Reply:
That logic is already in place from when we took the module. The if statement that was added was to prevent any simple items being shown in there that would lead to an empty set of items in the size up and size down feature. So for a configurable product where you can select size. I then see: 
![image](https://user-images.githubusercontent.com/3218936/109499226-83317a00-7a8c-11eb-94cc-5f78f2ccb0d3.png)

Reply:
Hi Minesh 

Thanks for clarifying. What I meant was that currently the logic shows all smaller sizes and all greater sizes than the current size. 

We want just the one size below and one size above. It was an oversight in the original implementation. Is that feasible? 

Cheers

Reply:
Hi James,
Just looking at this one so this one might be tricky as some clients don't have their sizes organised correct the order meaning, I'll need to take a look further into this one but it should seem possible of checking... If we are say getting a Size `M` return size `S` and `L` What logic would you want in place for Larger sizes say `L` when there is only `M and `S`?

Reply:
Hi James!

Just trying to follow up on this one the Size-Up and Size-Down text if that comes from the Modal itself which is part of the Toshi JS then all I need to worry about is passing the correct sizes as an array? And I take it Toshi should detect the rest the single size in place and change the text accordingly? 

Let me know if I'm following this correct.

Reply:
Hi Minesh, 

All you need to do is pass the array of sizes ordered by size. 

As we ideally just want 1 or two sizes in the array it will look like `['S', 'L']`. Assuming the customer chose the item in size M. 

The TOSHI code doesn't order the list itself, it will use it in whatever order it's provided in.

Reply:
Hey James, 
On this one what I've done made this logic happen in PHP so when its fetching sizes it will check the current size. And see if there are any available up and down (next and previous) within the array aka if I have as size `M` it will return `['S' ,'L']` 
While if I had say `XS` to `XL` and I select `XL` it would only provide me with `['L']`
![image](https://user-images.githubusercontent.com/3218936/111780483-629c5900-88af-11eb-80b7-7fe3efb73271.png)
See Image attached.

Reply:
Great, that is exactly the desired behaviour 👍